docs/specs/snp-genotyping-vector.md

```diff
@@ -0,0 +1,185 @@
+   ask for when doing discovery. Reducing the number of annotations
+   should reduce file size somewhat.
+
+2. I've set the '-contamination' option to zero (0.0) here rather than
+   0.05 we've used previously because I didn't realise that this
+   option down-samples only alternate alleles (I thought it
+   downsampled both alt and ref alleles), and so setting contamination
+   > 0 will create bias towards reference allele. We will deal with
+   contamination separately.
+
+3. When genotyping given alleles, GATK ignores any variant in the
+   known variants VCF that is not marked as PASS in the FILTER
+   field. However, we want to genotype at all variants in the known
+   variants VCF, regardless of what is in the FILTER column. Therefore
+   the known variants VCF must be created setting FILTER to PASS for
+   all variants to get this to work as desired.
+
+
+### Step: VCF to Zarr conversion
+
+Convert the VCF to Zarr format via the scikit-allel
+[vcf_to_zarr()](https://scikit-allel.readthedocs.io/en/stable/io.html#allel.vcf_to_zarr)
+function. Note the following implementation details:
+
+* Only the ``variants/MQ``, ``calldata/GT``, ``calldata/GQ`` and
+  ``calldata/AD`` fields are required in the output.
+
+* Data types should be as follows: ``variants/MQ``: i1 (single byte
+  signed integer), ``calldata/GT``: i1; ``calldata/GQ``: i1;
+  ``calldata/AD``: i2
+
+* Compressor should be ``zarr.Blosc(cname='zstd', clevel=5,
+  shuffle=-1)``
+
+* Chunk length should be ``2**18``
+
+* ``alt_number`` parameter should be 3 (fix number of alt alleles)
+
+* Data should be grouped by contig (chromosome). This means calling
+  ``vcf_to_zarr()`` once for each contig in the reference genome,
+  specifying the name of the contig as the value of the ``region`` and
+  ``group`` parameters.
+
+Here is an example of the appropriate function call in Python for a
+single contig (this should be called for all contigs in the genome):
+
+```python
+import sys
+import allel
+import zarr
+
+
+def build_sample_zarr(input_path, output_path, contig):
+    allel.vcf_to_zarr(
+        input=input_path,
+        output=output_path,
+        group=contig,
+        region=contig,
+        compressor=zarr.Blosc(cname='zstd', clevel=5, shuffle=0),
+        overwrite=True,
+        fields=['calldata/GT', 'calldata/GQ', 'calldata/AD', 'variants/MQ'],
+        types={'calldata/GT': 'i1',
+               'calldata/GQ': 'i1',
+               'calldata/AD': 'i2',
+	       'variants/MQ': 'i1'},
+        alt_number=3,
+        chunk_length=2**18,
+        log=sys.stdout,
+    )
+
+```
+
+
+### Step: Zip Zarr
+
+Once the zarr conversion is complete, create a zip archive from the
+contents of the root folder of the zarr directory. E.g., if
+``{sample}`` is the sample ID and ``{sample}.zarr`` is the name of the
+output directory used in calls to ``vcf_to_zarr()``, then do:
+
+```bash
+cd {sample}.zarr
+zip -rmT0 {sample}.zarr.zip .
```

Comment:
`-r` recurse paths
`-m` move, ie delete existing dir and contents
`-T` test, ie only delete on a successful zip. 
`-0` no compression.